JSON/testsuite/src/JSONTest.cpp

```diff
@@ -1933,7 +1933,7 @@ void JSONTest::testEscapeUnicode()
 	result = parser.parse(utf8Text);
 	object = result.extract<Object::Ptr>();
 	ss.str(""); object->stringify(ss);
-	assert (ss.str() == "{\"name\":\"g\\u00FCnter\"}");
+	assert (ss.str() == "{\"name\":\"g\xC3\xBCnter\"}");
```

Comment:
I'd leave this test as is and create another one that tests functionality without escaping. This change will break the code, so there should be a way to force the parser to escape.

Reply:
I disagree - the code was broken by 1.8.0, and this is returning it back to the former functionality. This test would fail as is with versions < 1.8.0

Reply:
Perhaps I did not state it correctly. This test is named `testEscapeUnicode` and it should do what its name implies - test (un)escaping the unicode. What was before does not matter, we should do the right thing now. And that includes allowing end user to choose whether the returned JSON string is unicode-escaped or not - probably just change in the JSON::stringifier.

Reply:
I'm not able to commit much more time on this; is there a small tweak to the PR that would mean you'd accept it? Perhaps renaming the test to `testParsingUnicode`, which is most of what it does anyway?

Reply:
That would be fine, but the main concern is that a json containing unicode characters can not be escaped when stringified (that does not look like a big change, either).

Reply:
They don't need to be escaped though - any compliant JSON parser must accept the non-escaped characters as well as the escaped ones, as this test is demonstrating.

Reply:
user puts in
```
"{ \"name\" : \"\\u4e2d\" }"
```
does something with it (add or remove something), and gets out
```
"{\"name\":\"\xE4\xB8\xAD\"}"
```
now, it is true that those things are equivalent for JSON and a compliant parser must accept them both. but from my experience, people can be more picky than parsers, especially about the superficial things. I'll finish it, thanks for reporting the issue and the fixes so far.

Reply:
Thanks very much, appreciate it!